Sources/LinkedIn/Core/HttpClient.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  HttpClient.swift
+//  PovioKitAuth
+//
+//  Created by Borut Tomazin on 22/05/2024.
+//  Copyright © 2024 Povio Inc. All rights reserved.
+//
+
+import Foundation
+
+struct HttpClient {
+  func request(method: String, url: URL, headers: [Header]) async throws -> Data {
+    var urlRequest = URLRequest(url: url)
+    urlRequest.httpMethod = method
+    headers.forEach {
```

Comment:
You can set `allHTTPHeaderFields` directly (the type of `headers` needs to be adjusted).

Reply:
What is the benefit here?

Reply:
Now, an array of tuples is created, traversed, and converted into a dictionary.
Instead, you can create a dictionary and directly assign it to the request. 🤷‍♂️

Reply:
https://github.com/poviolabs/PovioKitAuth/pull/30/commits/dd8a2bc491936eea0e76941dd04646702625f46b